### ISSUE TITLE
False negative for `Style/MethodCallWithoutArgsParentheses`

### ISSUE THREAD

**Issue by harto** [rubocop/rubocop#11467]
`Style/MethodCallWithoutArgsParentheses` doesn't report an issue when assigning to a variable with the same name as the method, e.g.:
```ruby
foo = x.foo()
```
It correctly reports an issue when the method and variable names are different:
```ruby
bar = x.foo()
```

This seems like an unintended side-effect of #2056.

## Expected behavior

Both of the above should report an issue.

## Actual behavior

Only the latter case reports an issue.

## Steps to reproduce the problem

```console
~/src % cat >test.rb
foo = x.foo()
bar = x.foo()
~/src % rubocop --only Style/MethodCallWithoutArgsParentheses test.rb 
Inspecting 1 file
C

Offenses:

test.rb:2:12: C: Style/MethodCallWithoutArgsParentheses: Do not use parentheses for method calls with no arguments.
bar = x.foo()
           ^^

1 file inspected, 1 offense detected
```

## RuboCop version

```console
~/src % rubocop -V       
0.88.0 (using Parser 3.1.0.0, rubocop-ast 0.2.0, running on ruby 2.7.7 arm64-darwin21)
```
